tracee-ebpf/tracee/tracee.bpf.c

```diff
@@ -2053,6 +2064,98 @@ static __always_inline unsigned short get_inode_mode_from_fd(u64 fd)
     return READ_KERN(f_inode->i_mode);
 }
 
+static __always_inline void detect_hooked_syscalls(event_data_t *data){
+    int key = 0;
+    u64 *table_ptr = bpf_map_lookup_elem(&syscall_table_addr_map, (void *)&key);
+    if (table_ptr == NULL)
+        return ;
+
+    u64 *start_ptr = bpf_map_lookup_elem(&code_boundaries_map, (void *)&key);
+    if (start_ptr == NULL)
+        return ;
+
+    key++;
+    u64 *end_ptr= bpf_map_lookup_elem(&code_boundaries_map, (void *)&key);
+    if (end_ptr == NULL)
+        return ;
+
+    unsigned long *syscall_table_addr = (unsigned long*) *table_ptr;
+    unsigned long stext = *start_ptr;
+    unsigned long etext = *end_ptr;
+
+    #if defined(bpf_target_x86)
+        int syscalls_to_check[] = {
+            0,          // read
+            1,          // write
+            2,          // open
+            3,          // close
+            16,         // ioctl
+            41,         // socket
+            59,         // execve
+            62,         // kill
+            78,         // getdents
+            101,        // ptrace
+            217,        // getdents64
+            321,        // bpf
+            322,        // execveat
+            322,        // openat
+            44,          // sendto
+            45,         // recvfrom
+            46,     	// sendmsg
+            47	        // recvmsg
+
+        };
+    #elif defined(bpf_target_arm64)
+        int syscalls_to_check[] = {
+            63,         // read
+            64,         // write
+                        // open (not present)
+            57,         // close
+            29,         // ioctl
+            198,        // socket
+            221,        // execve
+            129,        // kill
+                        // getdents (not present)
+            117,        // ptrace
+            61,         // getdents64
+            280,        // bpf
+            281,        // execveat
+            56,         // openat
+            206,        // sendto
+            207,        // recvfrom
+            211,        // sendmsg
+            212	        // recvmsg
+
+        };
+    #endif
+
+        int j=0;
+        int snum;                // syscall_number
+        unsigned long saddr = 0; // syscall address
+
+        #pragma unroll
+        for(int i = 0; i < sizeof(syscalls_to_check)/4; i++) {
+            snum = syscalls_to_check[i];
+            saddr = READ_KERN(syscall_table_addr[snum]);
+
+            if (saddr == 0)
+                continue;
+
+            // syscall address MUST BE inside kernel text segment boundaries
+
+            if(!((saddr >= stext) && (saddr < etext))) {
+                hooked_syscall_t hooked_syscall = {snum, saddr};
+                bpf_map_update_elem(&hooked_syscalls_map, &j, &hooked_syscall, BPF_ANY);
+                j++;
+            }
```

Comment:
I would take this logic to userspace as anyways you get the stext and etext from there (using kallsysms).
This way we can remove the need for code_boundaries_map and just send the syscalls addresses for ALL the configured syscalls. The check will then be performed in userspace.

Other than that, we can push all the addresses as an integer array argument (we should define a new type for that) and remove the need for hooked_syscalls_map as well. This is better than reading the map in userspace after receiving the event.

Reply:
Sounds good , ill change it that it will just "fetch" the asked syscalls to map and then we will parse it in the user-space.

About the integer array idea, I think its isn't will give us so much more then we already have because we still have to change the argument type tot a struct that will contain also the hooking module/object of of the syscall (see `HookedSyscallData`)